Updates for both Jubrunr and Spring Boot 3 (#303)
AxonIQ/reference-guide: axon-framework/deadlines/event-schedulers.md

 3. [Quartz](http://www.quartz-scheduler.org/) based
 4. [Axon Server](../../axon-server/introduction.md) based

The pure-Java implementation of the `EventScheduler` uses a `ScheduledExecutorService` to schedule event publication. 
Although the timing of this scheduler is very reliable, it is a pure in-memory implementation. 
Once the JVM is shut down, all schedules are lost. 
This makes this implementation unsuitable for long-term schedules.
The `SimpleEventScheduler` needs to be configured with an `EventBus` and a `SchedulingExecutorService` \(see the static methods on the `java.util.concurrent.Executors` class for helper methods\).

The `JonRunrEventScheduler` is a more reliable and enterprise-worthy implementation.
It offers several ways to persist the scheduled jobs, and has good integration with Spring Boot.
Using JobRunr as underlying scheduling mechanism, it provides more powerful features, such as clustering, misfire management as well as an optional dashboard.
This means event publication is guaranteed. It might be a little late, but it will be published.
It needs to be configured with a JobRunr `JobScheduler`, an `EventBus` and a `Serializer`.

> The `QuartzEventScheduler` is an alternative enterprise-worthy implementation, but the project has not much recent activity. 

Reviewer: ```suggestion
The `QuartzEventScheduler` is an alternative enterprise-worthy implementation, but the project has not seen much recent activity. 
```

>
> Spring Boot users which rely on Axon Server do not have to define anything.
> The auto configuration will automatically create a `AxonServerEventScheduler` for them.
> 
> Spring Boot users which want to use the JobRunr event scheduler can add [`jobrunr-spring-boot-starter`](https://mvnrepository.com/artifact/org.jobrunr/jobrunr-spring-boot-starter) as dependency.
> In addition adding a `EventScheduler` bean configuration needs to be added. This should look like:
> ```java
> @Bean
> public EventScheduler eventScheduler(
>         @Qualifier("eventSerializer") final Serializer serializer,
>         final JobScheduler jobScheduler,
>         final EventBus eventBus,
>         final TransactionManager transactionManager,
> ) {
>     return JobRunrEventScheduler.builder()
>             .jobScheduler(jobScheduler)
>             .serializer(serializer)
>             .eventBus(eventBus)

> >             .transactionManager(transactionManager)

Reviewer: Don't forget the `SpanFactory`!

>
> Spring Boot users which rely on Axon Server do not have to define anything.
> The auto configuration will automatically create a `AxonServerEventScheduler` for them.
> 

> > Spring Boot users which want to use the JobRunr event scheduler can add [`jobrunr-spring-boot-starter`](https://mvnrepository.com/artifact/org.jobrunr/jobrunr-spring-boot-starter) as dependency.
> > In addition adding a `EventScheduler` bean configuration needs to be added. This should look like:

Reviewer: ```suggestion
> Spring Boot users who want to use the JobRunr event scheduler can add [`jobrunr-spring-boot-starter`](https://mvnrepository.com/artifact/org.jobrunr/jobrunr-spring-boot-starter) as a dependency.
> In addition, an `EventScheduler` bean configuration needs to be added. This should look like:
```
